nexus/db-queries/src/db/datastore/physical_disk.rs

```diff
@@ -338,7 +341,7 @@ mod test {
             String::from("Noxide"),
             String::from("456"),
             String::from("UnrealDisk"),
-            PhysicalDiskKind::M2,
+            PhysicalDiskKind::U2,
```

Comment:
Changed this to a `U2` b/c the `M2` is filtered out in the results

Reply:
Personally I find this a little confusing, an internal function called `sled_list_physical_disks` doesn't indicate that it would filter out M.2s.

If we're keeping that name: we should continue inserting this M.2 in the test, and keep returning it.
Otherwise, we should rename `sled_list_physical_disks` to `sled_list_external_disks` or something.

Reply:
I'm fine w/ changing the name if that helps. We can't leave the M.2s in because `sled_list_physical_disk` pipes up to the app layer `sled_list_physical_disk` which maps to `/v1/system/hardware/sleds/:sled/disks`.

Reply:
Why wouldn't we just list both the U.2s and M.2s? The original reason we had confusion was that I think @ericaasen saw the wrong types here. This endpoint feels like it should list everything. We talked in #3482 about terminology challenges, but I don't see why wouldn't list all disks here.